src/test_driver/linux-cirque/helper/CHIPTestBase.py

```
@@ -0,0 +1,287 @@
+            if device.device_specification.device_type == 'wifi_ap'][0]
+
+        self.logger.info("wifi ap ssid: {}, psk: {}".format(ssid, psk))
+
+        for device in self.non_ap_devices:
+            self.logger.info(
+                "device id: {} connecting to desired ssid: {}".format(
+                    device.device_id, ssid))
+            write_psk_to_wpa_supplicant_config(
+                self.logger, self.stub, self.home_id, device.device_id, ssid, psk)
+
+            kill_existing_wpa_supplicant(
+                self.logger, self.stub, self.home_id, device.device_id)
+
+            start_wpa_supplicant(
+                self.logger, self.stub, self.home_id, device.device_id)
+        time.sleep(5)
+
+    def get_device_thread_ip(self, device_id):
+        ret = self.stub.ExecuteDeviceCommand(
+                service_pb2.ExecuteDeviceCommandRequest(
+                    home_id=self.home_id,
+                    device_id=device_id,
+                    command="ot-ctl ipaddr",
+                    streaming=False))
+        ipaddr_list = ret.output.splitlines()
+        for ipstr in ipaddr_list:
+            try:
+                self.logger.info("device id: {} thread ip: {}".format(device_id, ipstr))
+                ipaddr = ipaddress.ip_address(ipstr)
+                if ipaddr.is_link_local:
+                    self.logger.info("ignore ip: {} : link local".format(ipstr))
+                    continue
+                if not ipaddr.is_private:
+                    self.logger.info("ignore ip: {} : global ip addr".format(ipstr))
+                    continue
+                if re.match("fd[0-9a-f]{2}:[0-9a-f]{4}:[0-9a-f]{4}:[0-9a-f]{4}:0000:00ff:fe00:[0-9a-f]{4}", ipaddr.exploded) != None:
+                    self.logger.info("ignore ip: {} : not for app".format(ipstr))
+                    continue
+                self.logger.info("return ip: {}".format(ipstr))
+                return str(ipaddr)
+            except:
```

what exceptions can happen here? 

Could you add a comment about them and why it is appropriate to silently ignore them? I am used to at least logging exceptions.

---

Add a comment to explain why just ignore the exception